src/daemon/bitcoind/poller.rs

```diff
@@ -392,30 +399,75 @@ fn mark_confirmed_emers(
     Ok(())
 }
 
+enum ToBeCpfped {
+    Spend(SpendTransaction),
+    Unvault(UnvaultTransaction),
+}
+
+impl ToBeCpfped {
+    pub fn txid(&self) -> Txid {
+        match self {
+            Self::Spend(s) => s.txid(),
+            Self::Unvault(u) => u.txid(),
+        }
+    }
+
+    pub fn cpfp_txin(&self, der_desc: &DerivedCpfpDescriptor) -> Option<CpfpTxIn> {
+        match self {
+            Self::Spend(s) => s.cpfp_txin(der_desc),
+            Self::Unvault(u) => u.cpfp_txin(der_desc),
+        }
+    }
+
+    pub fn max_weight(&self) -> u64 {
+        match self {
+            Self::Spend(s) => s.max_weight(),
+            Self::Unvault(u) => u.max_weight(),
+        }
+    }
+
+    pub fn fees(&self) -> Amount {
+        // TODO(revault_tx): fees() should return an Amount!
+        Amount::from_sat(match self {
+            Self::Spend(s) => s.fees(),
+            Self::Unvault(u) => u.fees(),
+        })
+    }
+}
+
+// CPFP a bunch of transactions, bumping their feerate by at least `target_feerate`.
+// `target_feerate` is expressed in sat/kWU.
+// All the transactions' feerate MUST be below `target_feerate`.
 fn cpfp_package(
     revaultd: &Arc<RwLock<RevaultD>>,
     bitcoind: &BitcoinD,
-    mut tx_package: Vec<impl CpfpableTransaction>,
-    current_feerate: u64,
+    mut tx_package: Vec<ToBeCpfped>,
```

Comment:
Can we call this a package?

Reply:
Not really, i fix it elsewhere IIRC. If not here, the rationale was that i didn't want to increase the diff by too much.